projects/core/src/lib/layers/legend/legend-raster/raster-legend.component.ts

```diff
@@ -56,6 +56,52 @@ export class CastMeasurementToContinuousPipe implements PipeTransform {
     }
 }
 
+export function unifyDecimals(values: number[]): number[] {
+    // Early return if all values differ by more than 1
+    if (values.every((x) => x >= 1)) {
+        const toCompare: number[] = values.map((x) => Math.floor(x)).sort((a, b) => a - b);
+        let oneApart = true;
+        for (let i = 0; i < toCompare.length - 1; i++) {
+            if (toCompare[i + 1] - toCompare[i] < 1) {
+                oneApart = false;
+            }
+        }
+        if (oneApart) return values.map((x) => Math.floor(x));
+    }
+
+    // Find highest overlap for cut-off point
+    let maxOverlap = 0;
+    values.forEach((x) => {
+        values.forEach((y) => {
+            if (x === y) {
+                return;
+            }
+            const overlap: number = overlappingDigits(x, y);
+            maxOverlap = overlap > maxOverlap ? overlap : maxOverlap;
+        });
+    });
```

Comment:
This seems to be highly inefficient. Please just check the adjacent values in a single loop.

Reply:
Right, in hindsight, what I did there is pretty embarrassing. Not sure why I thought I had to compare every value with every other one. I condensed it to a single loop that checks the adjacent values.